src/DataFilterer.ts

```diff
@@ -83,6 +82,72 @@ export class DataFilterer {
         return [datasets, compVars, totals];
     }
 
+    calculateMean(filterOptions: DataFiltererOptions,
+                  impactData:    ImpactDataRow[],
+                  plotColours:   { [p: string] : string }): any[] {
+        // compare will always be year!
+
+        const filtData = this.filterByAll(filterOptions, impactData);
+
+        let meanVars = this.meanVariables(filterOptions.metric);
+        const top = meanVars.top;
+        const bottom = meanVars.bottom;
+        
+        const temp_top = this.filterByCompare(-1, filterOptions.compare, top, filtData);
+        const compVars_top: any[] = temp_top[1];
+        const filteredData_top: ImpactDataRow[] = temp_top[0];
+        // get an array of all the remaining disagg values
+        const aggVars_top: any[] = [...this.getUniqueVariables(-1, filterOptions.disagg, top, filteredData_top)];
+        const dataByAggregate_top = this.groupDataByDisaggAndThenCompare(filterOptions.compare, filterOptions.disagg, aggVars_top, filteredData_top);
+
+        const temp_bottom = this.filterByCompare(-1, filterOptions.compare, bottom, filtData);
+        const compVars_bottom: any[] = temp_bottom[1];
+        const filteredData_bottom: ImpactDataRow[] = temp_bottom[0];
+        // get an array of all the remaining disagg values
+        const aggVars_bottom: any[] = [...this.getUniqueVariables(-1, filterOptions.disagg, bottom, filteredData_bottom)];
+        const dataByAggregate_bottom = this.groupDataByDisaggAndThenCompare(filterOptions.compare, filterOptions.disagg, aggVars_bottom, filteredData_bottom);
+        
+
+        let datasets: FilteredRow[] = [];
+        for (let aggVar of aggVars_top) {
+            let summedMetricForDisagg: number[] = this.reduceSummary(dataByAggregate_top,
+                                                                     aggVar,
+                                                                     compVars_top,
+                                                                     top);
+            if (bottom != null) {
+                let summedMetricForDisagg_bottom: number[] = this.reduceSummary(dataByAggregate_bottom,
```

Comment:
I think the calculation of `dataByAggregate_bottom` could be moved inside this conditional scope